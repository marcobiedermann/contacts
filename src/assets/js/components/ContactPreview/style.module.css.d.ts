declare const styles: {
  readonly contact__name: string;
  readonly contact__link: string;
  readonly 'is-active': string;
};
export = styles;
