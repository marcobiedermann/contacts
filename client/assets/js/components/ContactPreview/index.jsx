import PropTypes from 'prop-types';
import React from 'react';
import { NavLink } from 'react-router-dom';
import * as routes from '../../constants/routes';
import styles from './style.css';

const ContactPreview = props => {
  const { id, name } = props;

  return (
    <div>
      <h2 className={styles.contact__name}>
        <NavLink
          to={`${routes.CONTACTS}/${id}`}
          className={styles.contact__link}
          activeClassName="is-active"
        >
          {name.firstName} {name.lastName}
        </NavLink>
      </h2>
    </div>
  );
};

ContactPreview.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.shape(),
};

ContactPreview.defaultProps = {
  name: null,
};

export default ContactPreview;
