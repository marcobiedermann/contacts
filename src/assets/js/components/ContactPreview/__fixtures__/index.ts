import { ContactPreviewProps } from '..';

const props: ContactPreviewProps = {
  id: '1',
  name: {
    firstName: 'John',
    lastName: 'Doe',
  },
};

export default props;
