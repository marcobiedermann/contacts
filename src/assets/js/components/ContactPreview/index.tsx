import React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './style.module.css';

export interface ContactPreviewProps {
  id: string;
  name: {
    firstName: string;
    lastName: string;
  };
}

function ContactPreview(props: ContactPreviewProps): JSX.Element {
  const { id, name } = props;

  return (
    <div>
      <h2 className={styles.contact__name}>
        <NavLink to={`/contacts/${id}`} className={styles.contact__link} activeClassName="is-active">
          {name.firstName} {name.lastName}
        </NavLink>
      </h2>
    </div>
  );
}

export default ContactPreview;
