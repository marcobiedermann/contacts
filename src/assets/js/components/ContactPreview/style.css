.contact__name {
  font-size: 1.125rem;
  margin: 0;
}

.contact__link {
  color: inherit;
  display: block;
  padding: 0.5em;
}

.contact__link.is-active {
  background-color: #d8eefa;
}
