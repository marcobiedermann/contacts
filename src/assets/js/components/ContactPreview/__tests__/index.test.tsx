import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import ContactPreview from '..';
import props from '../__fixtures__';

describe('contact preview component', () => {
  it('should render correctly', () => {
    expect.assertions(1);

    const { container } = render(
      <MemoryRouter>
        <ContactPreview {...props} />
      </MemoryRouter>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
