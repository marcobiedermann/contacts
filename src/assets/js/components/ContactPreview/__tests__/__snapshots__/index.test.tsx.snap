// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact preview component should render correctly 1`] = `
<div>
  <h2
    class="contact__name"
  >
    <a
      class="contact__link"
      href="/contacts/1"
    >
      John
       
      Doe
    </a>
  </h2>
</div>
`;
